ReactWindows/ReactNative/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,255 @@
+        public static ImportantForAccessibility GetImportantForAccessibilityAttached(UIElement element)
+        {
+            return (ImportantForAccessibility)element.GetValue(ImportantForAccessibilityAttachedProperty);
+        }
+
+        /// <summary>
+        /// Recursively sets AccessibilityView to <paramref name="accessibilityView"/> for <paramref name="parentAutomationPeer"/> children.
+        /// </summary>
+        /// <param name="parentAutomationPeer"></param>
+        /// <param name="accessibilityView"></param>
+        private static void SetChildrenAccessibilityView(AutomationPeer parentAutomationPeer, AccessibilityView accessibilityView)
+        {
+            if (parentAutomationPeer?.GetChildren() == null)
+            {
+                return;
+            }
+
+            foreach (AutomationPeer childAutomationPeer in parentAutomationPeer.GetChildren())
+            {
+                UIElement childUIElement = GetUIElementFromAutomationPeer(childAutomationPeer);
+                if (childUIElement != null)
+                {
+                    AutomationProperties.SetAccessibilityView(childUIElement, accessibilityView);
+                }
+                SetChildrenAccessibilityView(childAutomationPeer, accessibilityView);
+            }
+        }
+
+        /// <summary>
+        /// Returns true if in <paramref name="uiElementAutomationPeer"/> parents chain there is an UIElement with
+        /// ImportantForAccessibility value of Yes or NoHideDescendants. 'Yes' denotes that the element is visible
+        /// to the narrator as a container i.e. its children are 'hidden'.
+        /// </summary>
+        /// <param name="uiElementAutomationPeer"></param>
+        /// <returns></returns>
+        private static bool IsHiddenByAncestor(AutomationPeer uiElementAutomationPeer)
+        {
+            if (uiElementAutomationPeer == null)
+            {
+                return false;
+            }
+
+            bool result;
+            switch (uiElementAutomationPeer.Navigate(AutomationNavigationDirection.Parent))
+            {
+                case FrameworkElementAutomationPeer parentFEAutomationPeer:
+                    if (parentFEAutomationPeer.Owner != null)
+                    {
+                        var importantForAccessibilityAttached = GetImportantForAccessibilityAttached(parentFEAutomationPeer.Owner);
+                        if (importantForAccessibilityAttached == ImportantForAccessibility.Yes ||
+                            importantForAccessibilityAttached == ImportantForAccessibility.NoHideDescendants)
+                        {
+                            return true;
+                        }
+                    }
+                    result = IsHiddenByAncestor(parentFEAutomationPeer);
+                    break;
+                case ItemAutomationPeer parentIAutomationPeer:
+                    if (parentIAutomationPeer.Item is UIElement parentUIElement)
```

Comment:
>parentUIElement [](start = 64, length = 15)

Do we need to check whether this is null before using it?

Reply:
No, since 'is' evaluates to false for 'null' or if expression can't be converted.

---
In reply to: [164512847](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r164512847) [](ancestors = 164512847)